resources/user.go

```diff
@@ -0,0 +1,275 @@
+	return obj.BaseRes.Init() // call base init, b/c we're overriding
+}
+
+// Watch is the primary listener for this resource and it outputs events.
+func (obj *UserRes) Watch() error {
+
+	var err error
+	obj.recWatcher, err = recwatch.NewRecWatcher(file, false)
+	if err != nil {
+		return err
+	}
+	defer obj.recWatcher.Close()
+
+	// notify engine that we're running
+	if err := obj.Running(); err != nil {
+		return err // bubble up a NACK...
+	}
+
+	var send = false // send event?
+	var exit *error
+
+	for {
+		if obj.debug {
+			log.Printf("Watching: %s", file) // attempting to watch...
+		}
+
+		select {
+		case event, ok := <-obj.recWatcher.Events():
+			if !ok { // channel shutdown
+				return nil
+			}
+			if err := event.Error; err != nil {
+				return errwrap.Wrapf(err, "Unknown %s watcher error", obj)
+			}
+			if obj.debug { // don't access event.Body if event.Error isn't nil
+				log.Printf("%s: Event(%s): %v", obj, event.Body.Name, event.Body.Op)
+			}
+			send = true
+			obj.StateOK(false) // dirty
+
+		case event := <-obj.Events():
+			if exit, send = obj.ReadEvent(event); exit != nil {
+				return *exit // exit
+			}
+			//obj.StateOK(false) // dirty // these events don't invalidate state
+		}
+
+		// do all our event sending all together to avoid duplicate msgs
+		if send {
+			send = false
+			obj.Event()
+		}
+	}
+}
+
+// CheckApply method for User resource. Does nothing, returns happy!
+func (obj *UserRes) CheckApply(apply bool) (checkOK bool, err error) {
+	log.Printf("%s: CheckApply(%t)", obj, apply)
+
+	var exists = true
+
+	usr, err := user.Lookup(obj.GetName())
+	if err != nil {
+		if err == err.(user.UnknownUserError) {
+			log.Printf("the user: %s does not exist", obj.GetName())
+			exists = false
+		} else {
+			return false, fmt.Errorf("error looking up user")
+		}
+	}
+
+	if obj.AllowDuplicateUID == false {
+		existingUID, err := user.LookupId(strconv.Itoa(int(*obj.UID)))
+		if err == nil && existingUID.Username != obj.GetName() {
+			return false, fmt.Errorf("the uid: %d is already taken", *obj.UID)
+		}
+	}
+
+	if exists && obj.State == "exists" {
+		usercheck := true
+		if obj.GID != nil && strconv.Itoa(int(*obj.GID)) != usr.Gid {
+			usercheck = false
+		}
+		if obj.UID != nil && strconv.Itoa(int(*obj.UID)) != usr.Uid {
+			usercheck = false
+		}
+		if obj.HomeDir != "" && obj.HomeDir != usr.HomeDir {
+			usercheck = false
+		}
+		if usercheck {
+			return true, nil
+		}
+	}
+
+	if obj.State == "absent" && !exists {
+		return true, nil
+	}
+
+	if !apply {
+		return false, nil
+	}
+
+	var cmdName string
+	var args []string
+	if obj.State == "exists" {
+		if !exists {
+			cmdName = "useradd"
+			log.Printf("adding user: %s", obj.GetName())
+		}
+		if exists {
+			cmdName = "usermod"
+			log.Printf("modifying user: %s", obj.GetName())
+		}
+		if obj.AllowDuplicateUID == true {
+			args = append(args, "--non-unique")
+		}
+		if obj.UID != nil {
+			args = append(args, "-u", strconv.Itoa(int(*obj.UID)))
```

Comment:
instead of all the `strconv.Itoa(int(x)))` everywhere, you might want to consider: fmt.Sprintf("%d", x) instead.